### DISCUSSION TITLE
How do I create nested models without multiple collections when using for mongodb

### DISCUSSION THREAD

**Discussion by xenoterracide** [prisma/prisma#16778] [Q&A]
is there an example of doing mongo db with nested objects but not multiple collections? or can someone give me one?

**Comment by nurul3101** [comment]
Hello @xenoterracide 👋 
It looks like you are looking for [composite types](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-composite-types), can you have a look and let us know if that's what you need?

**Reply by xenoterracide** [reply]
looks like it, I'll give it a go. I'm guessing `@ignore` means I can't say don't store a field? that seems bad... is there a way to transform a field into a different structure, and give it a different name? I'm hoping yes. Not sure if I currently have an ignored field use case, but I think I do.

**Reply by xenoterracide** [reply]
actually I know I have a property I don't want to store. The timezone on an object, because that should be ephemeral, injected on load, coming from an external configuration.

**Reply by nurul3101** [reply]
You would need to add the field in the model in the schema file, even if the field is stored ephemerally it would still need to be defined in the model.

**Reply by xenoterracide** [reply]
are you then able to mark the field as ephemeral? I mean, I assume that's what `@ignore`? does but it's documented as not working

**Reply by nurul3101** [reply]
Yes, you can mark the fields as optional by adding a question mark(?)
`@ignore` - [reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#ignore) would just exclude the field from generated PrismaClient.

**Reply by xenoterracide** [reply]
but how does that say "don't store this field"? I assume, perhaps incorrectly, that optional simply means null/undefined

**Reply by xenoterracide** [reply]
hmm.... maybe a better question is, does prisma support a rich model? or would I basically embed it as a dao layer.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
